File: 310971/Guzik/task-320/src/main/java/com/example/entitiesapp/entities/Article.java
package com.example.entitiesapp.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "tbl_article")
@Getter
@Setter
public class Article extends BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "writer_id", nullable = false)
    private Writer writer;

    @Column(name = "title", nullable = false, length = 64)
    private String title;

    @Column(name = "content", nullable = false, length = 2048)
    private String content;

    @OneToMany(mappedBy = "article", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Post> posts = new ArrayList<>();

    @ManyToMany
    @JoinTable(
            name = "article_sticker",
            joinColumns = @JoinColumn(name = "article_id"),
            inverseJoinColumns = @JoinColumn(name = "sticker_id")
    )
    private Set<Sticker> stickers = new HashSet<>();
}

File: 351004/Kuchko/ServerApp/ServerApp/Models/Entities/Article.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ServerApp.Models.Entities;

[Table("tbl_article")]
public class Article : BaseEntity
{
    [Column("author_id")] public long AuthorId { get; set; }

    [ForeignKey("AuthorId")] public virtual Author Author { get; set; } = null!;

    [Column("title")] public string Title { get; set; } = null!;

    [Column("content")] public string Content { get; set; } = null!;

    [Column("created")] public DateTime Created { get; set; }

    [Column("modified")] public DateTime Modified { get; set; }

    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();
    public virtual ICollection<Sticker> Stickers { get; set; } = new List<Sticker>();
}